Lib/test/test_unicode.py

```diff
@@ -617,7 +618,14 @@ def test_isspace(self):
         self.checkequalnofix(True, '\u2000', 'isspace')
         self.checkequalnofix(True, '\u200a', 'isspace')
         self.checkequalnofix(False, '\u2014', 'isspace')
-        # apparently there are no non-BMP spaces chars in Unicode 6
+        for i in range(0x10000):
```

Comment:
I would prefer to have a constant declared at the top level of the module. Why not testing the full range of the UCS? range(0x110000)

Reply:
> I would prefer to have a constant declared at the top level of the module.

Ah, good idea. I think in fact perhaps I'll make the range itself a constant, and then be able to write:
```
for codepoint in BMP_CODEPOINTS:
  char = chr(codepoint)
```

because that seems to get at the meaning very well.

(Or perhaps `for char in bmp_chars():`. OTOH I just spent a few minutes trying to see a clean way to make it `for char in BMP_CHARS:`, and couldn't find one.)

Reply:
> Why not testing the full range of the UCS? range(0x110000)

Well, the most causal answer to "why" :slightly_smiling_face: is that I first added this as a test in `test_unicodedata.py` before seeing the existing `test_isspace` here -- and two of the central tests there (running through the whole database and computing overall checksums) have a loop much like this one, so I followed their lead.

Those two tests (`test_method_checksum` and `test_function_checksum`) look to be substantially unchanged since they were first added in 6a20ee7de back in 2000. They probably should be updated to go through the full range of codepoints; I think back in 2000 this was the full range of codepoints.

(Hmm... as I look closer at what `test_method_checksum` in particular does, it's really all about methods on `str`, and doesn't actually mention `unicodedata`. Perhaps it should move to `test_unicode.py`.)

The one real cost of looping through all planes (vs. just the BMP) is test runtime -- the test takes about 50ms for just the BMP (on my fast desktop) but 900ms for everything. Fast test suites are precious, so in `test_isspace` it seems a bit of a shame to spend almost a second looping through values where nothing at all is expected to happen. Well worth it for the loops in `test_unicodedata`, though, where there's plenty happening the whole way along.

Reply:
> I would prefer to have a constant declared at the top level of the module. Why not testing the full range of the UCS? range(0x110000)

Just pushed an update making both of these changes. Thanks for the review!

I was about to edit the existing two loops in `test_unicodedata` to run the full range too, but then realized that that requires updating the checksum. I feel like that should be a separate change and not mixed in with this one. So I'll keep that around as a commit on a branch on top of this one; I can send it right after this PR is complete, or also happy to squash it in here if you think that'd be preferable.